### PR TITLE
Jest stateful test

### PR DESCRIPTION
**Pull request by leongaban**
This is a pull request to help with the debugging of a non-passing stateful component test.

Question posted here: https://stackoverflow.com/questions/56637899/jest-stateful-component-with-async-function-failing-test

This PR is to get attention on my problem per the authors recommendation at the conclusion of the plurasight course here: https://app.pluralsight.com/player?course=testing-react-applications-jest&author=daniel-stern&name=f31785ac-30f9-4d02-aab6-473c85ce6770&clip=8&mode=live

### CODE REVIEW

File: src/components/NotificationsViewer.spec.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+      );
+
+    await delay();
+       
+    const instance = tree.root;
+    const component = instance.findByProps({className: `notifications`});
+    const text = component.children[0];
+    console.log('text is:', text);
+
+    expect(text).toEqual('42 Notifications Awaiting!');
```
Comment (leongaban) [inline src/components/NotificationsViewer.spec.js]: Problem is here, text is still stuck displaying `Loading...` even though the service has been mocked and count changed to `42`.

Comment (leongaban) [reply src/components/NotificationsViewer.spec.js]: This is now fixed! Using Enzyme and the following line, which will call the `componentDidMount` life cycle method.

`await wrapper.instance().componentDidMount();`

**Comment by JulianoPadilha** [comment]
In fact, the real problem was in `isomorphic-react/src/components/__tests__/NotificationsViewer.js` file. The `delay` are imported the wrong way and that was causing the error on test.

If importing `delay` like this: `import { delay } from 'redux-saga'` fix the problem. =D

**Comment by leongaban** [comment]
OH you are right!

I had: import delay from 'redux-saga';

However I think me going down this route was actually pretty helpful in the
long run, since a lot of projects don't use or rely on redux-saga yet, it's
good to figure out a way to use Enzyme in this case to make sure that
componentDidMount was called.

**Pull request merged**